diff --git a/node.gyp b/node.gyp
index 08cb3f3..16bacc9 100644
--- a/node.gyp
+++ b/node.gyp
@@ -83,7 +83,7 @@
       [ 'OS=="win" and '
         'node_use_openssl=="true" and '
         'node_shared_openssl=="false"', {
-        'use_openssl_def%': 1,
+        'use_openssl_def%': 0,
       }, {
         'use_openssl_def%': 0,
       }],
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 4fca0eb..5974e2d 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -781,6 +781,11 @@
         '<@(inspector_all_sources)',
       ],
       'conditions': [
+        ['component=="shared_library"', {
+          'defines': [
+            'BUILDING_V8_SHARED',
+          ],
+        }],
         ['v8_enable_third_party_heap==1', {
           # TODO(targos): add values from v8_third_party_heap_files to sources
         }, {
diff --git a/deps/v8/src/heap/factory-base.h b/deps/v8/src/heap/factory-base.h
index 53e83a8..793d2f9 100644
--- a/deps/v8/src/heap/factory-base.h
+++ b/deps/v8/src/heap/factory-base.h
@@ -62,5 +62,4 @@ class EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE) TorqueGeneratedFactory {
 template <typename Impl>
-class EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE) FactoryBase
-    : public TorqueGeneratedFactory<Impl> {
+class FactoryBase : public TorqueGeneratedFactory<Impl> {
  public:
   // Converts the given boolean condition to JavaScript boolean value.
@@ -318,5 +317,9 @@ class EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE) FactoryBase
   friend TorqueGeneratedFactory<Impl>;
 };
-
+extern template class EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
+    FactoryBase<Factory>;
+extern template class EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
+    FactoryBase<LocalFactory>;
+
 }  // namespace internal
 }  // namespace v8
